### PR TITLE
scylladb: init at 3.0.5

### PR DESCRIPTION
**Pull request by workflow**
###### Motivation for this change

Initial build of [ScyllaDB](https://www.scylladb.com/), the real-time big data database.

This is an MVP, let me know what should be cleaned up.

Warning: [Long build ahead](https://en.wikipedia.org/wiki/Scylla)!

cc: @dingxiangfei2009 @snawaz

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)

- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/libantlr3cpp/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{stdenv, fetchgit}:
+
+stdenv.mkDerivation {
+  name = "libantlr3cpp-3.5";
+  src = fetchgit {
+    url = "https://github.com/antlr/antlr3.git";
```
Comment (Infinisil) [inline pkgs/development/libraries/libantlr3cpp/default.nix]: Use `fetchFromGitHub` for github repos

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -1658,6 +1658,11 @@
     github = "fare";
     name = "Francois-Rene Rideau";
   };
+  farlion = {
+    email = "florian.peter@gmx.at";
+    github = "workflow";
+    name = "Florian Peter";
+  };
```
Comment (Infinisil) [inline maintainers/maintainer-list.nix]: Separate this into a different commit, same PR

File: pkgs/servers/scylladb/seastar-configure-script-paths.patch [hunk 3]
```diff
@@ -0,0 +1,13 @@
+diff --git a/seastar/configure.py b/seastar/configure.py
+index 62d9c204..f6520635 100755
+--- a/seastar/configure.py
++++ b/seastar/configure.py
+@@ -924,7 +924,7 @@ with open(buildfile, 'w') as f:
+             command = ragel -G2 -o $out $in && sed -i -e '1h;2,$$H;$$!d;g' -re 's/static const char _nfa[^;]*;//g' $out
+             description = RAGEL $out
+         rule gen
+-            command = /bin/echo -e $text > $out
++            command = echo -e $text > $out
```
Comment (Infinisil) [inline pkgs/servers/scylladb/seastar-configure-script-paths.patch]: I can't find /bin/echo in the upstream source, is this patch really needed?

File: pkgs/servers/scylladb/default.nix [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  buildPhase = ''
+    ninja -j "$NIX_BUILD_CORES" -v
+  '';
+  installPhase = ''
+    mkdir $out
+    cp -r * $out/
+  '';
+  meta = with stdenv.lib; {
+    description = "NoSQL data store using the seastar framework, compatible with Apache Cassandra";
+    homepage = http://scylladb.com;
```
Comment (Infinisil) [inline pkgs/servers/scylladb/default.nix]: Also supports HTTPS

File: pkgs/servers/scylladb/default.nix [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+  };
+
+  patches = [ ./seastar-configure-script-paths.patch ];
+
+  nativeBuildInputs = [ pkgconfig ];
+
+  buildInputs = [
+   python3Packages.pyparsing
+   python3
+   gcc8
```
Comment (Infinisil) [inline pkgs/servers/scylladb/default.nix]: Use `gcc8Stdenv` for `stdenv` instead of adding `gcc8` like this

File: pkgs/servers/scylladb/default.nix [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+   boost
+   git
+   systemd
+   gnutls
+   cmake
+   makeWrapper
+   ninja
```
Comment (Infinisil) [inline pkgs/servers/scylladb/default.nix]: Tools that are only needed at build time like ninja, makeWrapper, etc. should go into `nativeBuildInputs`

File: pkgs/servers/scylladb/default.nix [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+let
+  thriftNixpkgs = builtins.fetchGit {
+    name = "thrift-0.10";
+    url = https://github.com/nixos/nixpkgs/;
+    rev = "fbaa12bad90e3c8726a5427fdd981dc60c7f08ff";
+  };
+  thriftPkgs = import thriftNixpkgs {};
+in
+stdenv.mkDerivation rec {
+  name = "scylladb-${version}";
```
Comment (Infinisil) [inline pkgs/servers/scylladb/default.nix]: Use `pname = "scylladb"` instead

**Review by infinisil — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 2 above)
Comment (workflow) [reply maintainers/maintainer-list.nix]: Done, thanks

File: pkgs/servers/scylladb/default.nix [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+  snappy,
+  libtool
+}:
+let
+  thriftNixpkgs = builtins.fetchGit {
+    name = "thrift-0.10";
+    url = https://github.com/nixos/nixpkgs/;
+    rev = "fbaa12bad90e3c8726a5427fdd981dc60c7f08ff";
+  };
+  thriftPkgs = import thriftNixpkgs {};
```
[the remark this answers is not in the document]
Comment (workflow) [reply pkgs/servers/scylladb/default.nix]: Got it, the last version of `thrift` that scylla seems to compile with is `0.10`, so I added that to `master` nixpkgs. Is this approach legit?

File: pkgs/development/libraries/libantlr3cpp/default.nix (hunk 1 above)
Comment (workflow) [reply pkgs/development/libraries/libantlr3cpp/default.nix]: Nice, thanks.

File: pkgs/servers/scylladb/seastar-configure-script-paths.patch (hunk 3 above)
Comment (workflow) [reply pkgs/servers/scylladb/seastar-configure-script-paths.patch]: Unfortunately, yep, it's well hidden because of git submodules: https://github.com/scylladb/scylla-seastar/blob/f541231a3011420e26fb33e71355415334dcb861/configure.py#L927

File: pkgs/servers/scylladb/default.nix (hunk 5 above)
Comment (workflow) [reply pkgs/servers/scylladb/default.nix]: Cool, didn't know that!

File: pkgs/servers/scylladb/default.nix (hunk 6 above)
Comment (workflow) [reply pkgs/servers/scylladb/default.nix]: Will try to filter them out. Any straightforward way to do this automatically?

File: pkgs/servers/scylladb/default.nix (hunk 7 above)
Comment (workflow) [reply pkgs/servers/scylladb/default.nix]: done

**Comment by workflow** [comment]
@Infinisil Addressed all comments. Thank you very much for your review!

File: pkgs/development/libraries/libantlr3cpp/default.nix [hunk 9]
```diff
@@ -0,0 +1,25 @@
+{stdenv, fetchFromGitHub}:
+
+stdenv.mkDerivation {
+  name = "libantlr3cpp-3.5";
```
Comment (aanderse) [inline pkgs/development/libraries/libantlr3cpp/default.nix]: `pname` and `version` please.

File: pkgs/development/libraries/libantlr3cpp/default.nix [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+  dontBuild = true;
+
+  installPhase = ''
+    mkdir -p $out/include
+    cp runtime/Cpp/include/* $out/include/
+  '';
+
+  meta = with stdenv.lib; {
+    description = "C++ runtime libraries of ANTLR v3";
```
Comment (aanderse) [inline pkgs/development/libraries/libantlr3cpp/default.nix]: Given the `installPhase` you have provided this sounds misleading. What is the difference between this package and `antlr`?

File: pkgs/development/libraries/libantlr3cpp/default.nix [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+    homepage = http://www.antlr3.org/;
```
Comment (aanderse) [inline pkgs/development/libraries/libantlr3cpp/default.nix]: Please quote.

File: pkgs/development/libraries/libantlr3cpp/default.nix [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+  };
+
+  dontBuild = true;
+
+  installPhase = ''
+    mkdir -p $out/include
+    cp runtime/Cpp/include/* $out/include/
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (aanderse) [inline pkgs/development/libraries/libantlr3cpp/default.nix]: Missing `maintainers`.

File: pkgs/development/libraries/libsystemtap/default.nix [hunk 13]
```diff
@@ -0,0 +1,27 @@
+{stdenv, fetchgit, gettext, python, elfutils}:
+
+stdenv.mkDerivation {
+  name = "libsystemtap";
```
Comment (aanderse) [inline pkgs/development/libraries/libsystemtap/default.nix]: `pname` and `version` please.

File: pkgs/development/libraries/libsystemtap/default.nix [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+  src = fetchgit {
+    url = git://sourceware.org/git/systemtap.git;
+    rev = "4051c70c9318c837981384cbb23f3e9eb1bd0892";
+    sha256 = "0sd8n3j3rishks3gyqj2jyqhps7hmlfjyz8i0w8v98cczhhh04rq";
+    fetchSubmodules = false;
+  };
+
+  dontBuild = true;
+
+  nativeBuildInputs = [ gettext python elfutils ];
```
Comment (aanderse) [inline pkgs/development/libraries/libsystemtap/default.nix]: Given `dontBuild` is set to true are all of these used?

File: pkgs/development/libraries/libsystemtap/default.nix [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+
+  installPhase = ''
+    mkdir -p $out/include
+    cp -r includes/* $out/include/
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (aanderse) [inline pkgs/development/libraries/libsystemtap/default.nix]: `maintainers` please.

File: pkgs/development/libraries/libsystemtap/default.nix [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+    description = "systemtap-sdt-dev - statically defined probes development files";
```
Comment (aanderse) [inline pkgs/development/libraries/libsystemtap/default.nix]: Please see `nixpkgs` manual for appropriate `description` values.

File: pkgs/development/libraries/libsystemtap/default.nix [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+    homepage = https://sourceware.org/systemtap/;
```
Comment (aanderse) [inline pkgs/development/libraries/libsystemtap/default.nix]: Please quote.

File: pkgs/development/libraries/thrift/0.10.nix [hunk 18]
```diff
@@ -0,0 +1,39 @@
+{ stdenv, fetchurl, boost, zlib, libevent, openssl, python, pkgconfig, bison
+, flex, twisted
+}:
+
+stdenv.mkDerivation rec {
+  name = "thrift-${version}";
```
Comment (aanderse) [inline pkgs/development/libraries/thrift/0.10.nix]: Please replace `name` with `pname = "thrift";` as `name` is already defined elsewhere.

File: pkgs/development/libraries/thrift/0.10.nix [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+
+  #enableParallelBuilding = true; problems on hydra
+
+  # Workaround to make the python wrapper not drop this package:
+  # pythonFull.buildEnv.override { extraLibs = [ thrift ]; }
+  pythonPath = [];
+
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs = [
+    boost zlib libevent openssl python bison flex twisted
```
Comment (aanderse) [inline pkgs/development/libraries/thrift/0.10.nix]: Are `bison` and `flex` actually used after building the application? My guess is they belong in `nativeBuildInputs`.

File: pkgs/development/libraries/thrift/0.10.nix [hunk 20]
```diff
@@ -0,0 +1,39 @@
...
+  preConfigure = "export PY_PREFIX=$out";
+
+  # TODO: package boost-test, so we can run the test suite. (Currently it fails
+  # to find libboost_unit_test_framework.a.)
+  configureFlags = [ "--enable-tests=no" ];
+  doCheck = false;
+
+  meta = with stdenv.lib; {
+    description = "Library for scalable cross-language services";
+    homepage = http://thrift.apache.org/;
```
Comment (aanderse) [inline pkgs/development/libraries/thrift/0.10.nix]: Please quote.

File: pkgs/servers/scylladb/default.nix [hunk 21]
```diff
@@ -0,0 +1,100 @@
...
+   libyamlcpp
+   libsystemtap
+   lksctp-tools
+   lz4
+   libxml2
+   zlib
+   libpciaccess
+   snappy
+   libtool
+   thrift-0_10
```
Comment (aanderse) [inline pkgs/servers/scylladb/default.nix]: Provably best to just accept `thrift` here and then override the version is `all-packages.nix`.

File: pkgs/servers/scylladb/default.nix [hunk 22]
```diff
@@ -0,0 +1,100 @@
...
+  buildPhase = ''
+    ninja -j "$NIX_BUILD_CORES" -v
+  '';
+  installPhase = ''
+    mkdir $out
+    cp -r * $out/
+  '';
+  meta = with stdenv.lib; {
+    description = "NoSQL data store using the seastar framework, compatible with Apache Cassandra";
+    homepage = https://scylladb.com;
```
Comment (aanderse) [inline pkgs/servers/scylladb/default.nix]: Please quote.

File: pkgs/servers/scylladb/default.nix [hunk 23]
```diff
@@ -0,0 +1,100 @@
...
+   libxml2
+   zlib
+   libpciaccess
+   snappy
+   libtool
+   thrift-0_10
+  ];
+
+  configurePhase = ''
+     patchShebangs ./configure.py
```
Comment (aanderse) [inline pkgs/servers/scylladb/default.nix]: Maybe move this to the `prePatchPhase` or at least `preConfigure`. The main point being don't override `configurePhase` as that can cause problems and skip steps.

File: pkgs/servers/scylladb/default.nix [hunk 24]
```diff
@@ -0,0 +1,100 @@
...
+    ./configure.py --mode=release
```
Comment (aanderse) [inline pkgs/servers/scylladb/default.nix]: How about using `configureFlags` instead.

File: pkgs/servers/scylladb/default.nix [hunk 25]
```diff
@@ -0,0 +1,100 @@
...
+  '';
+  buildPhase = ''
+    ninja -j "$NIX_BUILD_CORES" -v
```
Comment (aanderse) [inline pkgs/servers/scylladb/default.nix]: I thought `ninja` build "just work" with `nix`. Any specific reason for this?

**Review by aanderse — CHANGES_REQUESTED** [review]
I've listed a few changes I hope you'll find helpful.

File: pkgs/development/libraries/libsystemtap/default.nix [hunk 26]
```diff
@@ -0,0 +1,27 @@
+{stdenv, fetchgit, gettext, python, elfutils}:
```
Comment (FRidh) [inline pkgs/development/libraries/libsystemtap/default.nix]: can we use python3?

File: pkgs/development/libraries/thrift/0.10.nix [hunk 27]
```diff
@@ -0,0 +1,39 @@
...
+
+  src = fetchurl {
+    url = "https://archive.apache.org/dist/thrift/${version}/${name}.tar.gz";
+    sha256 = "02x1xw0l669idkn6xww39j60kqxzcbmim4mvpb5h9nz8wqnx1292";
+  };
+
+  #enableParallelBuilding = true; problems on hydra
+
+  # Workaround to make the python wrapper not drop this package:
+  # pythonFull.buildEnv.override { extraLibs = [ thrift ]; }
```
Comment (FRidh) [inline pkgs/development/libraries/thrift/0.10.nix]: That's not how it should be done. Apparently this provides Python bindings. Those bindings should be made available via `python-packages.nix`.

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/libsystemtap/default.nix (hunk 13 above)
Comment (workflow) [reply pkgs/development/libraries/libsystemtap/default.nix]: Couldn't find a version on the project, so calling it `1.0.0.`

File: pkgs/development/libraries/libantlr3cpp/default.nix (hunk 10 above)
Comment (workflow) [reply pkgs/development/libraries/libantlr3cpp/default.nix]: Good question! For some reason, even though the `antlr` package provides libraries, these are currently at version `2.7.7` which is incompatible with `3.x`, and the existing `antlr3` package does not include any libraries.

Maybe I'm missing something though?

File: pkgs/development/libraries/libsystemtap/default.nix (hunk 14 above)
Comment (workflow) [reply pkgs/development/libraries/libsystemtap/default.nix]: Yap, all three are being used in the `configure` phase. Any better place to put them?

File: pkgs/development/libraries/libsystemtap/default.nix (hunk 16 above)
Comment (workflow) [reply pkgs/development/libraries/libsystemtap/default.nix]: Thanks, improved :)

File: pkgs/development/libraries/thrift/0.10.nix (hunk 19 above)
Comment (workflow) [reply pkgs/development/libraries/thrift/0.10.nix]: I'm following the example of the original `thrift` package in version 0.10 here.

Happy to improve that package, but maybe in a separate PR?

File: pkgs/servers/scylladb/default.nix (hunk 21 above)
Comment (workflow) [reply pkgs/servers/scylladb/default.nix]: That's a very good idea, done thanks :)

File: pkgs/development/libraries/thrift/0.10.nix (hunk 19 above)
Comment (aanderse) [reply pkgs/development/libraries/thrift/0.10.nix]: Separate PR sounds good. Thanks for flipping the extra mile on that!

File: pkgs/development/libraries/libsystemtap/default.nix (hunk 14 above)
Comment (aanderse) [reply pkgs/development/libraries/libsystemtap/default.nix]: Ah sorry didn't catch that. Please ignore my comment, you're spot on.

File: pkgs/servers/scylladb/default.nix (hunk 23 above)
Comment (workflow) [reply pkgs/servers/scylladb/default.nix]: Thanks! Turns out I don't need this at all ^__^ removed it.

File: pkgs/development/libraries/libantlr3cpp/default.nix (hunk 10 above)
Comment (aanderse) [reply pkgs/development/libraries/libantlr3cpp/default.nix]: Oh I see... If I had to guess and if I trusted my memory on the issue (which I definitely do not :smile:) I would wonder if when`antlr3` was packaged for nixpkgs back in 2016 maybe the c++ support in `antlr3` wasn't that good so it was just skipped... :man_shrugging:

Maybe we should just amend the existing package?

File: pkgs/development/libraries/libsystemtap/default.nix (hunk 13 above)
Comment (aanderse) [reply pkgs/development/libraries/libsystemtap/default.nix]: **If** there is no stable release the convention is to set the version to `unstable-YYYY-MM-DD` where the date is the date of the commit you're referencing.

File: pkgs/servers/scylladb/default.nix (hunk 24 above)
Comment (workflow) [reply pkgs/servers/scylladb/default.nix]: Great point! 

What I'm trying to do, instead of overriding the `configurePhase`, is this:

```
  configureScript = "./configure.py";
  configureFlags = [ "--mode=release" ];
  dontAddPrefix = true;
```

However for some reason, whenever `cmake` is available as a `nativeBuildInput`, it ignores my overrides and tries to use `cmake` instead.

File: pkgs/servers/scylladb/default.nix (hunk 25 above)
Comment (workflow) [reply pkgs/servers/scylladb/default.nix]: You are absolutely Sir, it does work out of the box! Removed the `buildPhase` override entirely.

File: pkgs/development/libraries/libantlr3cpp/default.nix (hunk 10 above)
Comment (workflow) [reply pkgs/development/libraries/libantlr3cpp/default.nix]: Okay good point, I'll look into that!

File: pkgs/development/libraries/libsystemtap/default.nix (hunk 13 above)
Comment (workflow) [reply pkgs/development/libraries/libsystemtap/default.nix]: Good to know! It is actually tagged as `release-3.2` in the git repo, I was just being lazy.

Comment (workflow) [reply pkgs/development/libraries/libsystemtap/default.nix]: Updated version.

File: pkgs/servers/scylladb/default.nix (hunk 24 above)
Comment (workflow) [reply pkgs/servers/scylladb/default.nix]: To be more precise, it adds a `fixing cmake files...` files, which I can't seam to turn off and somehow leads to this:

```
configuring
fixing cmake files...
cmake flags: -GNinja -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON -DBUILD_TESTING=OFF -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/gccy6z2y06s0k8z
k743rar66438v0y7q-scylladb-3.0.5/include -DCMAKE_INSTALL_LIBDIR=/nix/store/gccy6z2y06s0k8zk743rar66438v0y7q-scylladb-3.0.5/lib -DCMAKE_INSTALL_NAME_DIR=/nix/store/gccy6z2y06s0k8zk743rar66438v0y7q-scylladb-3.0.5/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_DEPLOYMENT_TARGET= -
DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=last -DCMAKE_STRIP=/nix/store/6k95q6vvnmspf3c0nm8q55sixsvmn9nc-binutils-2.31.1/bin/strip -DCMAKE_RANLIB=/nix/store/6k95q6vvnmspf3c0nm8q55sixsvmn9nc-binutils-2.31.1/bin/ranlib -DCMAKE_AR=/nix/store/6k95q6vvnmspf3c0nm8q55sixsvmn9nc-binutils-
2.31.1/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++  
-- The C compiler identification is GNU 8.3.0
-- The CXX compiler identification is GNU 8.3.0
-- Check for working C compiler: /nix/store/lfhwv8kc8c94ix3y7bn34zz4k1c4zgha-gcc-wrapper-8.3.0/bin/gcc
-- Check for working C compiler: /nix/store/lfhwv8kc8c94ix3y7bn34zz4k1c4zgha-gcc-wrapper-8.3.0/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /nix/store/lfhwv8kc8c94ix3y7bn34zz4k1c4zgha-gcc-wrapper-8.3.0/bin/g++
-- Check for working CXX compiler: /nix/store/lfhwv8kc8c94ix3y7bn34zz4k1c4zgha-gcc-wrapper-8.3.0/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:9 (message):
  This CMakeLists.txt file is only valid for use in IDEs, please define
  FOR_IDE to acknowledge this.


-- Configuring incomplete, errors occurred!

```

File: pkgs/development/libraries/libantlr3cpp/default.nix (hunk 10 above)
Comment (workflow) [reply pkgs/development/libraries/libantlr3cpp/default.nix]: Amended the existing package instead! Let me know if that looks okay, please.

File: pkgs/servers/scylladb/default.nix (hunk 23 above)
Comment (workflow) [reply pkgs/servers/scylladb/default.nix]: Turns out `patchShebangs` *is* needed, I just happened to have the binary on my system and no sandboxing. Moved this into the `postPatch` phase as you recommended.

**Comment by workflow** [comment]
@aanderse Thanks for your wonderfully detailed review! Learned a lot, and addressed all comments I believe.

File: pkgs/development/libraries/libsystemtap/default.nix (hunk 26 above)
Comment (workflow) [reply pkgs/development/libraries/libsystemtap/default.nix]: Unfortunately, doesn't look like it.

`configure: error: python version 2 is required`

File: pkgs/development/libraries/thrift/0.10.nix (hunk 27 above)
Comment (workflow) [reply pkgs/development/libraries/thrift/0.10.nix]: Thanks for pointing that out!

Would you be okay if I look at that in a separate PR?

All I did here with `thrift` here is to reactivate the existing `0.10` thrift from `nixpkgs`, it's the vanilla `thrift 0.10` so to speak.

File: pkgs/servers/scylladb/default.nix [hunk 28]
```diff
@@ -0,0 +1,100 @@
+{ stdenv,
+  fetchgit,
+  python3Packages,
+  python3,
```
[the remark this answers is not in the document]
Comment (workflow) [reply pkgs/servers/scylladb/default.nix]: Thank you good point, removed `python3`.

**Comment by workflow** [comment]
@FRidh Thanks for your review as well! Please see my comments.

**Comment by aanderse** [comment]
@workflow Great work! How about you squash this PR into 2 commits: adding yourself to list off maintainers and the package init?

**Comment by aanderse** [comment]
ping @bjornfor @Infinisil @FRidh for next steps on this.

**Review by infinisil — APPROVED** [review]
Ideally a separate commit for every package. Other than that this looks good to merge to me, nice work :)

**Comment by Infinisil** [comment]
@GrahamcOfBorg build scylladb

**Comment by workflow** [comment]
Hmm CI seems to fail, but I can't find any useful output from the borgs. Will verify building it on nixos, traveling right now and won't have access to my remote builder until Tuesday.

It builds fine here on my Ubuntu box, but that doesn't really help without sandboxing.

**Comment by workflow** [comment]
@Infinisil Done rebasing, thanks! Tried the build on a couple of machines, including NixOS, and it builds fine (as long as it doesn't run out of memory, which can happen when using a lot of cores)

**Comment by workflow** [comment]
@GrahamcOfBorg build scylladb

File: pkgs/development/tools/parsing/antlr/3.5.nix [hunk 29]
```diff
@@ -1,18 +1,23 @@
-{stdenv, fetchurl, jre}:
+{stdenv, fetchurl, fetchFromGitHub, jre}:
 
 stdenv.mkDerivation rec {
-  name = "antlr-${version}";
+  pname = "antlr";
   version = "3.5.2";
-  src = fetchurl {
-    url ="https://www.antlr3.org/download/antlr-${version}-complete.jar";
+  jar = fetchurl {
```
Comment (Infinisil) [inline pkgs/development/tools/parsing/antlr/3.5.nix]: What's up with these antlr changes?

**Review by infinisil — COMMENTED** [review]
Commits are still a bit messed up. The first one is good, but the second one includes parts that should be in others. Each commit should be self-contained, this means for new packages this should include the addition in all-packages.nix.

File: pkgs/development/tools/parsing/antlr/3.5.nix (hunk 29 above)
Comment (workflow) [reply pkgs/development/tools/parsing/antlr/3.5.nix]: This includes the antlr libraries in the package output, which are not currently present in `antlr 3.5.nix`

My initial approach was to add a new package `libantlr3cpp` that exposes the libs only, and @aanderse [suggested to modify the existing package](https://github.com/NixOS/nixpkgs/pull/61438#discussion_r306251757) instead!

**Comment by workflow** [comment]
Done and rebased.

@Infinisil quite an attention to detail :)

**Comment by Infinisil** [comment]
Hehe yeah, I kind of hate to do it, but it's better for the long-term quality of nixpkgs, since people learn from it and can do better PR's from the start in the future

That said, I have another complaint: The commit ordering needs to be adjusted, because in the current ordering, `thrift-0_10` is used in the second commit, when it's only introduced in the 5th one (meaning scylladb will fail to even evaluate in the second commit).

So the ordering should be something like
- maintainer-list
- libsystemtap, thrift and antlr
- scylladb

**Comment by workflow** [comment]
Haha, I actually noticed that as I rebased, and hoped to get away with it ;)
Not with this eagle-eye.

Thank you @Infinisil, appreciated!

Reordered.

**Comment by workflow** [comment]
Weird, Github still shows the old order here, but it's correctly reordered downstream: https://github.com/tenx-tech/nixpkgs/commits/scylladb-upstream

**Comment by Infinisil** [comment]
Yeah GitHub unfortunately has [this bug](https://help.github.com/en/articles/why-are-my-commits-in-the-wrong-order) (No it's not a feature, GitHub!)

I can see the order is correct with `git log`, thanks :)

**Review by infinisil — APPROVED** [review]
